src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -859,7 +863,8 @@ static bool tcp_do_read(grpc_tcp* tcp, grpc_error_handle* error)
 
 static void maybe_make_read_slices(grpc_tcp* tcp)
     ABSL_EXCLUSIVE_LOCKS_REQUIRED(tcp->read_mu) {
-  if (tcp->incoming_buffer->length == 0 &&
```

Comment:
Let's separate this piece out into another PR: the API change will be a complicated cherry-pick, and this change may need to roll back - so by splitting them we give ourselves the luxury of an easy rollback if needed - without the need to orchestrate a second cherry-pick.

Reply:
Done